opengever/maintenance/scripts/fix_retention_period.py

```diff
@@ -0,0 +1,433 @@
+                                self.item['_query_path'],
+                                self.current_period,
+                                self.new_period))
+        self.make_retention_period_backup(self.context)
+        ILifeCycle(self.context).retention_period = self.new_period
+        return True
+
+    def apply_to_dossier(self, dossier):
+        dossier_path = '/'.join(dossier.getPhysicalPath())
+        current_period = ILifeCycle(dossier).retention_period
+
+        if current_period != self.current_period:
+            if self.options.verbose:
+                logger.info('skipping dossier {}, modified'
+                            .format(dossier_path))
+            return
+
+        if current_period == self.new_period:
+            if self.options.verbose:
+                logger.info('skipping dossier {}, not changed'
+                            .format(dossier_path))
+            return
+
+        if self.options.verbose:
+            logger.info('fixing dossier {}, {}->{}'
+                        .format(dossier_path, current_period, self.new_period))
+
+        ILifeCycle(dossier).retention_period = self.new_period
+
+        # the log entry is untranslated on purpose, we don't want to add
+        # translations for this fix to opengever.core.
+        comment = u'Alter Wert: "{} Jahre", neuer Wert: "{} Jahre"'.format(
+            current_period, self.new_period)
+        journal_entry_factory(dossier, 'Aufbewahrungsdauer korrigiert',
+                              title=u'Aufbewahrungsdauer korrigiert.',
+                              comment=comment)
+
+
+class RetentionPeriodFixer(XlsSource):
+    """Fix retention periods that were not set from the xls file by mistake.
+
+    The problem was caused by a missing row-header in the init excel file.
+    Thus all retention_period values specified by the customer were not set,
+    instead the default value was used.
+
+    We attempt to fix this by comparing the current repository to the initial
+    excel file and setting the retention_period on the repository as follows:
+
+                    plone: no change, plone: (re-)moved, plone: changed
+    xls: no value   parent/default,   skip,              skip
+    xls: value      from xls,         skip,              skip
+
+    Values specified in the excel file are inherited by children. If an excel
+    file contanis a value for a parent folder but none for a child folder the
+    child folder inherits the parent's retention period setting.
+
+    When a leaf folder's retention_period is updated, all dossiers in that
+    folder that have the same retention_period as the folder are updated
+    as well, under the assumption that the value was inherited.
+
+    Note: this will set a wrong retention_period when:
+      - the repo_folders retention_period changes to non-default
+      - the child (dossier/repo-folder) should have a rentention_period equal
+        to the default retention_period.
```

Comment:
`to the value of what was previously a default, but was a deliberately chosen value for that dossier`